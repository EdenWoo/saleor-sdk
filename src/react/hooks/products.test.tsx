import { act, renderHook } from "@testing-library/react-hooks";
import React from "react";
import { ApolloProvider } from "react-apollo";
import setupAPI from "../../../testUtils/api";
import { useProductList } from "./products";
import { OrderDirection, ProductOrderField } from "../../gqlTypes/globalTypes";

const { client } = setupAPI();

const wrapper: React.FC = ({ children }) => (
  <ApolloProvider client={client}>{children}</ApolloProvider>
);

describe("useProductList", () => {
  it("can fetch products", async () => {
    const { result } = renderHook(
      () =>
        useProductList({
          first: 5,
        }),
      {
        wrapper,
      }
    );

    expect(result.current.data).toBe(undefined);
    expect(result.current.loading).toBe(true);

    // @ts-ignore
    await act(() => result.current.current);

    expect(result.current.data).toMatchSnapshot();
    expect(result.current.loading).toBe(false);
  });

  it("can fetch next page", async () => {
    const { result } = renderHook(
      () =>
        useProductList({
          first: 1,
        }),
      {
        wrapper,
      }
    );

    // @ts-ignore
    await act(() => result.current.current);

    expect(result.current.data).toMatchSnapshot();
    expect(result.current.data).toHaveLength(1);

    await act(result.current.next);

    expect(result.current.data).toMatchSnapshot();
    expect(result.current.loading).toBe(false);
    expect(result.current.data).toHaveLength(2);
  });

  it("can sort", async () => {
    const { result } = renderHook(
      () =>
        useProductList({
          first: 5,
          sortBy: {
            direction: OrderDirection.DESC,
            field: ProductOrderField.NAME,
          },
        }),
      {
        wrapper,
      }
    );

    // @ts-ignore
    await act(() => result.current.current);

    expect(result.current.data).toMatchSnapshot();
  });

  it("can filter", async () => {
    const { result } = renderHook(
      () =>
        useProductList({
          filter: {
            search: "beer",
          },
          first: 2,
        }),
      {
        wrapper,
      }
    );

    // @ts-ignore
    await act(() => result.current.current);

    expect(result.current.data).toMatchSnapshot();
  });
});
