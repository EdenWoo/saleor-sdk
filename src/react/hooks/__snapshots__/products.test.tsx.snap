// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useProductList can fetch next page 1`] = `
Array [
  Object {
    "__typename": "Product",
    "id": "UHJvZHVjdDo3Mg==",
    "name": "Apple Juice",
    "pricing": Object {
      "__typename": "ProductPricingInfo",
      "onSale": true,
      "priceRange": Object {
        "__typename": "TaxedMoneyRange",
        "start": Object {
          "__typename": "TaxedMoney",
          "gross": Object {
            "__typename": "Money",
            "amount": 1.5,
            "currency": "USD",
          },
          "net": Object {
            "__typename": "Money",
            "amount": 1.5,
            "currency": "USD",
          },
        },
        "stop": Object {
          "__typename": "TaxedMoney",
          "gross": Object {
            "__typename": "Money",
            "amount": 3.5,
            "currency": "USD",
          },
          "net": Object {
            "__typename": "Money",
            "amount": 3.5,
            "currency": "USD",
          },
        },
      },
      "priceRangeUndiscounted": Object {
        "__typename": "TaxedMoneyRange",
        "start": Object {
          "__typename": "TaxedMoney",
          "gross": Object {
            "__typename": "Money",
            "amount": 3,
            "currency": "USD",
          },
          "net": Object {
            "__typename": "Money",
            "amount": 3,
            "currency": "USD",
          },
        },
        "stop": Object {
          "__typename": "TaxedMoney",
          "gross": Object {
            "__typename": "Money",
            "amount": 7,
            "currency": "USD",
          },
          "net": Object {
            "__typename": "Money",
            "amount": 7,
            "currency": "USD",
          },
        },
      },
    },
    "thumbnail": Object {
      "__typename": "Image",
      "alt": "",
      "url": "http://localhost:8000/media/__sized__/products/saleordemoproduct_fd_juice_06_rNQ3p2L-thumbnail-255x255.png",
    },
    "thumbnail2x": Object {
      "__typename": "Image",
      "url": "http://localhost:8000/media/__sized__/products/saleordemoproduct_fd_juice_06_rNQ3p2L-thumbnail-510x510.png",
    },
  },
]
`;

exports[`useProductList can fetch next page 2`] = `
Array [
  Object {
    "__typename": "Product",
    "id": "UHJvZHVjdDo3Mg==",
    "name": "Apple Juice",
    "pricing": Object {
      "__typename": "ProductPricingInfo",
      "onSale": true,
      "priceRange": Object {
        "__typename": "TaxedMoneyRange",
        "start": Object {
          "__typename": "TaxedMoney",
          "gross": Object {
            "__typename": "Money",
            "amount": 1.5,
            "currency": "USD",
          },
          "net": Object {
            "__typename": "Money",
            "amount": 1.5,
            "currency": "USD",
          },
        },
        "stop": Object {
          "__typename": "TaxedMoney",
          "gross": Object {
            "__typename": "Money",
            "amount": 3.5,
            "currency": "USD",
          },
          "net": Object {
            "__typename": "Money",
            "amount": 3.5,
            "currency": "USD",
          },
        },
      },
      "priceRangeUndiscounted": Object {
        "__typename": "TaxedMoneyRange",
        "start": Object {
          "__typename": "TaxedMoney",
          "gross": Object {
            "__typename": "Money",
            "amount": 3,
            "currency": "USD",
          },
          "net": Object {
            "__typename": "Money",
            "amount": 3,
            "currency": "USD",
          },
        },
        "stop": Object {
          "__typename": "TaxedMoney",
          "gross": Object {
            "__typename": "Money",
            "amount": 7,
            "currency": "USD",
          },
          "net": Object {
            "__typename": "Money",
            "amount": 7,
            "currency": "USD",
          },
        },
      },
    },
    "thumbnail": Object {
      "__typename": "Image",
      "alt": "",
      "url": "http://localhost:8000/media/__sized__/products/saleordemoproduct_fd_juice_06_rNQ3p2L-thumbnail-255x255.png",
    },
    "thumbnail2x": Object {
      "__typename": "Image",
      "url": "http://localhost:8000/media/__sized__/products/saleordemoproduct_fd_juice_06_rNQ3p2L-thumbnail-510x510.png",
    },
  },
  Object {
    "__typename": "Product",
    "id": "UHJvZHVjdDo3NA==",
    "name": "Banana Juice",
    "pricing": Object {
      "__typename": "ProductPricingInfo",
      "onSale": true,
      "priceRange": Object {
        "__typename": "TaxedMoneyRange",
        "start": Object {
          "__typename": "TaxedMoney",
          "gross": Object {
            "__typename": "Money",
            "amount": 2.1,
            "currency": "USD",
          },
          "net": Object {
            "__typename": "Money",
            "amount": 2.1,
            "currency": "USD",
          },
        },
        "stop": Object {
          "__typename": "TaxedMoney",
          "gross": Object {
            "__typename": "Money",
            "amount": 4.9,
            "currency": "USD",
          },
          "net": Object {
            "__typename": "Money",
            "amount": 4.9,
            "currency": "USD",
          },
        },
      },
      "priceRangeUndiscounted": Object {
        "__typename": "TaxedMoneyRange",
        "start": Object {
          "__typename": "TaxedMoney",
          "gross": Object {
            "__typename": "Money",
            "amount": 3,
            "currency": "USD",
          },
          "net": Object {
            "__typename": "Money",
            "amount": 3,
            "currency": "USD",
          },
        },
        "stop": Object {
          "__typename": "TaxedMoney",
          "gross": Object {
            "__typename": "Money",
            "amount": 7,
            "currency": "USD",
          },
          "net": Object {
            "__typename": "Money",
            "amount": 7,
            "currency": "USD",
          },
        },
      },
    },
    "thumbnail": Object {
      "__typename": "Image",
      "alt": "",
      "url": "http://localhost:8000/media/__sized__/products/saleordemoproduct_fd_juice_01_J9ZIGUm-thumbnail-255x255.png",
    },
    "thumbnail2x": Object {
      "__typename": "Image",
      "url": "http://localhost:8000/media/__sized__/products/saleordemoproduct_fd_juice_01_J9ZIGUm-thumbnail-510x510.png",
    },
  },
]
`;

exports[`useProductList can fetch products 1`] = `
Array [
  Object {
    "__typename": "Product",
    "id": "UHJvZHVjdDo3Mg==",
    "name": "Apple Juice",
    "pricing": Object {
      "__typename": "ProductPricingInfo",
      "onSale": true,
      "priceRange": Object {
        "__typename": "TaxedMoneyRange",
        "start": Object {
          "__typename": "TaxedMoney",
          "gross": Object {
            "__typename": "Money",
            "amount": 1.5,
            "currency": "USD",
          },
          "net": Object {
            "__typename": "Money",
            "amount": 1.5,
            "currency": "USD",
          },
        },
        "stop": Object {
          "__typename": "TaxedMoney",
          "gross": Object {
            "__typename": "Money",
            "amount": 3.5,
            "currency": "USD",
          },
          "net": Object {
            "__typename": "Money",
            "amount": 3.5,
            "currency": "USD",
          },
        },
      },
      "priceRangeUndiscounted": Object {
        "__typename": "TaxedMoneyRange",
        "start": Object {
          "__typename": "TaxedMoney",
          "gross": Object {
            "__typename": "Money",
            "amount": 3,
            "currency": "USD",
          },
          "net": Object {
            "__typename": "Money",
            "amount": 3,
            "currency": "USD",
          },
        },
        "stop": Object {
          "__typename": "TaxedMoney",
          "gross": Object {
            "__typename": "Money",
            "amount": 7,
            "currency": "USD",
          },
          "net": Object {
            "__typename": "Money",
            "amount": 7,
            "currency": "USD",
          },
        },
      },
    },
    "thumbnail": Object {
      "__typename": "Image",
      "alt": "",
      "url": "http://localhost:8000/media/__sized__/products/saleordemoproduct_fd_juice_06_rNQ3p2L-thumbnail-255x255.png",
    },
    "thumbnail2x": Object {
      "__typename": "Image",
      "url": "http://localhost:8000/media/__sized__/products/saleordemoproduct_fd_juice_06_rNQ3p2L-thumbnail-510x510.png",
    },
  },
  Object {
    "__typename": "Product",
    "id": "UHJvZHVjdDo3NA==",
    "name": "Banana Juice",
    "pricing": Object {
      "__typename": "ProductPricingInfo",
      "onSale": true,
      "priceRange": Object {
        "__typename": "TaxedMoneyRange",
        "start": Object {
          "__typename": "TaxedMoney",
          "gross": Object {
            "__typename": "Money",
            "amount": 2.1,
            "currency": "USD",
          },
          "net": Object {
            "__typename": "Money",
            "amount": 2.1,
            "currency": "USD",
          },
        },
        "stop": Object {
          "__typename": "TaxedMoney",
          "gross": Object {
            "__typename": "Money",
            "amount": 4.9,
            "currency": "USD",
          },
          "net": Object {
            "__typename": "Money",
            "amount": 4.9,
            "currency": "USD",
          },
        },
      },
      "priceRangeUndiscounted": Object {
        "__typename": "TaxedMoneyRange",
        "start": Object {
          "__typename": "TaxedMoney",
          "gross": Object {
            "__typename": "Money",
            "amount": 3,
            "currency": "USD",
          },
          "net": Object {
            "__typename": "Money",
            "amount": 3,
            "currency": "USD",
          },
        },
        "stop": Object {
          "__typename": "TaxedMoney",
          "gross": Object {
            "__typename": "Money",
            "amount": 7,
            "currency": "USD",
          },
          "net": Object {
            "__typename": "Money",
            "amount": 7,
            "currency": "USD",
          },
        },
      },
    },
    "thumbnail": Object {
      "__typename": "Image",
      "alt": "",
      "url": "http://localhost:8000/media/__sized__/products/saleordemoproduct_fd_juice_01_J9ZIGUm-thumbnail-255x255.png",
    },
    "thumbnail2x": Object {
      "__typename": "Image",
      "url": "http://localhost:8000/media/__sized__/products/saleordemoproduct_fd_juice_01_J9ZIGUm-thumbnail-510x510.png",
    },
  },
  Object {
    "__typename": "Product",
    "id": "UHJvZHVjdDo3OQ==",
    "name": "Bean Juice",
    "pricing": Object {
      "__typename": "ProductPricingInfo",
      "onSale": false,
      "priceRange": Object {
        "__typename": "TaxedMoneyRange",
        "start": Object {
          "__typename": "TaxedMoney",
          "gross": Object {
            "__typename": "Money",
            "amount": 3,
            "currency": "USD",
          },
          "net": Object {
            "__typename": "Money",
            "amount": 3,
            "currency": "USD",
          },
        },
        "stop": Object {
          "__typename": "TaxedMoney",
          "gross": Object {
            "__typename": "Money",
            "amount": 7,
            "currency": "USD",
          },
          "net": Object {
            "__typename": "Money",
            "amount": 7,
            "currency": "USD",
          },
        },
      },
      "priceRangeUndiscounted": Object {
        "__typename": "TaxedMoneyRange",
        "start": Object {
          "__typename": "TaxedMoney",
          "gross": Object {
            "__typename": "Money",
            "amount": 3,
            "currency": "USD",
          },
          "net": Object {
            "__typename": "Money",
            "amount": 3,
            "currency": "USD",
          },
        },
        "stop": Object {
          "__typename": "TaxedMoney",
          "gross": Object {
            "__typename": "Money",
            "amount": 7,
            "currency": "USD",
          },
          "net": Object {
            "__typename": "Money",
            "amount": 7,
            "currency": "USD",
          },
        },
      },
    },
    "thumbnail": Object {
      "__typename": "Image",
      "alt": "",
      "url": "http://localhost:8000/media/__sized__/products/saleordemoproduct_fd_juice_02_dNXPoKS-thumbnail-255x255.png",
    },
    "thumbnail2x": Object {
      "__typename": "Image",
      "url": "http://localhost:8000/media/__sized__/products/saleordemoproduct_fd_juice_02_dNXPoKS-thumbnail-510x510.png",
    },
  },
  Object {
    "__typename": "Product",
    "id": "UHJvZHVjdDoxMTU=",
    "name": "Black Hoodie",
    "pricing": Object {
      "__typename": "ProductPricingInfo",
      "onSale": false,
      "priceRange": Object {
        "__typename": "TaxedMoneyRange",
        "start": Object {
          "__typename": "TaxedMoney",
          "gross": Object {
            "__typename": "Money",
            "amount": 30,
            "currency": "USD",
          },
          "net": Object {
            "__typename": "Money",
            "amount": 30,
            "currency": "USD",
          },
        },
        "stop": Object {
          "__typename": "TaxedMoney",
          "gross": Object {
            "__typename": "Money",
            "amount": 30,
            "currency": "USD",
          },
          "net": Object {
            "__typename": "Money",
            "amount": 30,
            "currency": "USD",
          },
        },
      },
      "priceRangeUndiscounted": Object {
        "__typename": "TaxedMoneyRange",
        "start": Object {
          "__typename": "TaxedMoney",
          "gross": Object {
            "__typename": "Money",
            "amount": 30,
            "currency": "USD",
          },
          "net": Object {
            "__typename": "Money",
            "amount": 30,
            "currency": "USD",
          },
        },
        "stop": Object {
          "__typename": "TaxedMoney",
          "gross": Object {
            "__typename": "Money",
            "amount": 30,
            "currency": "USD",
          },
          "net": Object {
            "__typename": "Money",
            "amount": 30,
            "currency": "USD",
          },
        },
      },
    },
    "thumbnail": Object {
      "__typename": "Image",
      "alt": "",
      "url": "http://localhost:8000/media/__sized__/products/saleordemoproduct_cl_bogo01_1_7OtJd0E-thumbnail-255x255.png",
    },
    "thumbnail2x": Object {
      "__typename": "Image",
      "url": "http://localhost:8000/media/__sized__/products/saleordemoproduct_cl_bogo01_1_7OtJd0E-thumbnail-510x510.png",
    },
  },
  Object {
    "__typename": "Product",
    "id": "UHJvZHVjdDoxMTY=",
    "name": "Blue Hoodie",
    "pricing": Object {
      "__typename": "ProductPricingInfo",
      "onSale": false,
      "priceRange": Object {
        "__typename": "TaxedMoneyRange",
        "start": Object {
          "__typename": "TaxedMoney",
          "gross": Object {
            "__typename": "Money",
            "amount": 30,
            "currency": "USD",
          },
          "net": Object {
            "__typename": "Money",
            "amount": 30,
            "currency": "USD",
          },
        },
        "stop": Object {
          "__typename": "TaxedMoney",
          "gross": Object {
            "__typename": "Money",
            "amount": 30,
            "currency": "USD",
          },
          "net": Object {
            "__typename": "Money",
            "amount": 30,
            "currency": "USD",
          },
        },
      },
      "priceRangeUndiscounted": Object {
        "__typename": "TaxedMoneyRange",
        "start": Object {
          "__typename": "TaxedMoney",
          "gross": Object {
            "__typename": "Money",
            "amount": 30,
            "currency": "USD",
          },
          "net": Object {
            "__typename": "Money",
            "amount": 30,
            "currency": "USD",
          },
        },
        "stop": Object {
          "__typename": "TaxedMoney",
          "gross": Object {
            "__typename": "Money",
            "amount": 30,
            "currency": "USD",
          },
          "net": Object {
            "__typename": "Money",
            "amount": 30,
            "currency": "USD",
          },
        },
      },
    },
    "thumbnail": Object {
      "__typename": "Image",
      "alt": "",
      "url": "http://localhost:8000/media/__sized__/products/saleordemoproduct_cl_bogo02_1_mtuuFZd-thumbnail-255x255.png",
    },
    "thumbnail2x": Object {
      "__typename": "Image",
      "url": "http://localhost:8000/media/__sized__/products/saleordemoproduct_cl_bogo02_1_mtuuFZd-thumbnail-510x510.png",
    },
  },
]
`;

exports[`useProductList can filter 1`] = `
Array [
  Object {
    "__typename": "Product",
    "id": "UHJvZHVjdDo4NA==",
    "name": "Seaman Beer",
    "pricing": Object {
      "__typename": "ProductPricingInfo",
      "onSale": false,
      "priceRange": Object {
        "__typename": "TaxedMoneyRange",
        "start": Object {
          "__typename": "TaxedMoney",
          "gross": Object {
            "__typename": "Money",
            "amount": 3.2,
            "currency": "USD",
          },
          "net": Object {
            "__typename": "Money",
            "amount": 3.2,
            "currency": "USD",
          },
        },
        "stop": Object {
          "__typename": "TaxedMoney",
          "gross": Object {
            "__typename": "Money",
            "amount": 3.2,
            "currency": "USD",
          },
          "net": Object {
            "__typename": "Money",
            "amount": 3.2,
            "currency": "USD",
          },
        },
      },
      "priceRangeUndiscounted": Object {
        "__typename": "TaxedMoneyRange",
        "start": Object {
          "__typename": "TaxedMoney",
          "gross": Object {
            "__typename": "Money",
            "amount": 3.2,
            "currency": "USD",
          },
          "net": Object {
            "__typename": "Money",
            "amount": 3.2,
            "currency": "USD",
          },
        },
        "stop": Object {
          "__typename": "TaxedMoney",
          "gross": Object {
            "__typename": "Money",
            "amount": 3.2,
            "currency": "USD",
          },
          "net": Object {
            "__typename": "Money",
            "amount": 3.2,
            "currency": "USD",
          },
        },
      },
    },
    "thumbnail": Object {
      "__typename": "Image",
      "alt": "",
      "url": "http://localhost:8000/media/__sized__/products/saleordemoproduct_beer-01_1_Xm3ASfo-thumbnail-255x255.png",
    },
    "thumbnail2x": Object {
      "__typename": "Image",
      "url": "http://localhost:8000/media/__sized__/products/saleordemoproduct_beer-01_1_Xm3ASfo-thumbnail-510x510.png",
    },
  },
]
`;

exports[`useProductList can sort 1`] = `
Array [
  Object {
    "__typename": "Product",
    "id": "UHJvZHVjdDo4Nw==",
    "name": "Yellow Plimsolls",
    "pricing": Object {
      "__typename": "ProductPricingInfo",
      "onSale": false,
      "priceRange": Object {
        "__typename": "TaxedMoneyRange",
        "start": Object {
          "__typename": "TaxedMoney",
          "gross": Object {
            "__typename": "Money",
            "amount": 59,
            "currency": "USD",
          },
          "net": Object {
            "__typename": "Money",
            "amount": 59,
            "currency": "USD",
          },
        },
        "stop": Object {
          "__typename": "TaxedMoney",
          "gross": Object {
            "__typename": "Money",
            "amount": 59,
            "currency": "USD",
          },
          "net": Object {
            "__typename": "Money",
            "amount": 59,
            "currency": "USD",
          },
        },
      },
      "priceRangeUndiscounted": Object {
        "__typename": "TaxedMoneyRange",
        "start": Object {
          "__typename": "TaxedMoney",
          "gross": Object {
            "__typename": "Money",
            "amount": 59,
            "currency": "USD",
          },
          "net": Object {
            "__typename": "Money",
            "amount": 59,
            "currency": "USD",
          },
        },
        "stop": Object {
          "__typename": "TaxedMoney",
          "gross": Object {
            "__typename": "Money",
            "amount": 59,
            "currency": "USD",
          },
          "net": Object {
            "__typename": "Money",
            "amount": 59,
            "currency": "USD",
          },
        },
      },
    },
    "thumbnail": Object {
      "__typename": "Image",
      "alt": "",
      "url": "http://localhost:8000/media/__sized__/products/saleordemoproduct_sneakers_01_1_hxdEA8I-thumbnail-255x255.png",
    },
    "thumbnail2x": Object {
      "__typename": "Image",
      "url": "http://localhost:8000/media/__sized__/products/saleordemoproduct_sneakers_01_1_hxdEA8I-thumbnail-510x510.png",
    },
  },
  Object {
    "__typename": "Product",
    "id": "UHJvZHVjdDo4Mg==",
    "name": "White Wine",
    "pricing": Object {
      "__typename": "ProductPricingInfo",
      "onSale": true,
      "priceRange": Object {
        "__typename": "TaxedMoneyRange",
        "start": Object {
          "__typename": "TaxedMoney",
          "gross": Object {
            "__typename": "Money",
            "amount": 10.5,
            "currency": "USD",
          },
          "net": Object {
            "__typename": "Money",
            "amount": 10.5,
            "currency": "USD",
          },
        },
        "stop": Object {
          "__typename": "TaxedMoney",
          "gross": Object {
            "__typename": "Money",
            "amount": 10.5,
            "currency": "USD",
          },
          "net": Object {
            "__typename": "Money",
            "amount": 10.5,
            "currency": "USD",
          },
        },
      },
      "priceRangeUndiscounted": Object {
        "__typename": "TaxedMoneyRange",
        "start": Object {
          "__typename": "TaxedMoney",
          "gross": Object {
            "__typename": "Money",
            "amount": 15,
            "currency": "USD",
          },
          "net": Object {
            "__typename": "Money",
            "amount": 15,
            "currency": "USD",
          },
        },
        "stop": Object {
          "__typename": "TaxedMoney",
          "gross": Object {
            "__typename": "Money",
            "amount": 15,
            "currency": "USD",
          },
          "net": Object {
            "__typename": "Money",
            "amount": 15,
            "currency": "USD",
          },
        },
      },
    },
    "thumbnail": Object {
      "__typename": "Image",
      "alt": "",
      "url": "http://localhost:8000/media/__sized__/products/saleordemoproduct_wine-white_2TRD3SA-thumbnail-255x255.png",
    },
    "thumbnail2x": Object {
      "__typename": "Image",
      "url": "http://localhost:8000/media/__sized__/products/saleordemoproduct_wine-white_2TRD3SA-thumbnail-510x510.png",
    },
  },
  Object {
    "__typename": "Product",
    "id": "UHJvZHVjdDo4OA==",
    "name": "White Plimsolls",
    "pricing": Object {
      "__typename": "ProductPricingInfo",
      "onSale": true,
      "priceRange": Object {
        "__typename": "TaxedMoneyRange",
        "start": Object {
          "__typename": "TaxedMoney",
          "gross": Object {
            "__typename": "Money",
            "amount": 34.3,
            "currency": "USD",
          },
          "net": Object {
            "__typename": "Money",
            "amount": 34.3,
            "currency": "USD",
          },
        },
        "stop": Object {
          "__typename": "TaxedMoney",
          "gross": Object {
            "__typename": "Money",
            "amount": 34.3,
            "currency": "USD",
          },
          "net": Object {
            "__typename": "Money",
            "amount": 34.3,
            "currency": "USD",
          },
        },
      },
      "priceRangeUndiscounted": Object {
        "__typename": "TaxedMoneyRange",
        "start": Object {
          "__typename": "TaxedMoney",
          "gross": Object {
            "__typename": "Money",
            "amount": 49,
            "currency": "USD",
          },
          "net": Object {
            "__typename": "Money",
            "amount": 49,
            "currency": "USD",
          },
        },
        "stop": Object {
          "__typename": "TaxedMoney",
          "gross": Object {
            "__typename": "Money",
            "amount": 49,
            "currency": "USD",
          },
          "net": Object {
            "__typename": "Money",
            "amount": 49,
            "currency": "USD",
          },
        },
      },
    },
    "thumbnail": Object {
      "__typename": "Image",
      "alt": "",
      "url": "http://localhost:8000/media/__sized__/products/saleordemoproduct_sneakers_02_1_LKgUR6b-thumbnail-255x255.png",
    },
    "thumbnail2x": Object {
      "__typename": "Image",
      "url": "http://localhost:8000/media/__sized__/products/saleordemoproduct_sneakers_02_1_LKgUR6b-thumbnail-510x510.png",
    },
  },
  Object {
    "__typename": "Product",
    "id": "UHJvZHVjdDo4Ng==",
    "name": "White Parrot Cushion",
    "pricing": Object {
      "__typename": "ProductPricingInfo",
      "onSale": true,
      "priceRange": Object {
        "__typename": "TaxedMoneyRange",
        "start": Object {
          "__typename": "TaxedMoney",
          "gross": Object {
            "__typename": "Money",
            "amount": 10,
            "currency": "USD",
          },
          "net": Object {
            "__typename": "Money",
            "amount": 10,
            "currency": "USD",
          },
        },
        "stop": Object {
          "__typename": "TaxedMoney",
          "gross": Object {
            "__typename": "Money",
            "amount": 10,
            "currency": "USD",
          },
          "net": Object {
            "__typename": "Money",
            "amount": 10,
            "currency": "USD",
          },
        },
      },
      "priceRangeUndiscounted": Object {
        "__typename": "TaxedMoneyRange",
        "start": Object {
          "__typename": "TaxedMoney",
          "gross": Object {
            "__typename": "Money",
            "amount": 20,
            "currency": "USD",
          },
          "net": Object {
            "__typename": "Money",
            "amount": 20,
            "currency": "USD",
          },
        },
        "stop": Object {
          "__typename": "TaxedMoney",
          "gross": Object {
            "__typename": "Money",
            "amount": 20,
            "currency": "USD",
          },
          "net": Object {
            "__typename": "Money",
            "amount": 20,
            "currency": "USD",
          },
        },
      },
    },
    "thumbnail": Object {
      "__typename": "Image",
      "alt": "",
      "url": "http://localhost:8000/media/__sized__/products/saleordemoproduct_cuschion02_2rlBRlJ-thumbnail-255x255.png",
    },
    "thumbnail2x": Object {
      "__typename": "Image",
      "url": "http://localhost:8000/media/__sized__/products/saleordemoproduct_cuschion02_2rlBRlJ-thumbnail-510x510.png",
    },
  },
  Object {
    "__typename": "Product",
    "id": "UHJvZHVjdDoxMTg=",
    "name": "White Hoodie",
    "pricing": Object {
      "__typename": "ProductPricingInfo",
      "onSale": true,
      "priceRange": Object {
        "__typename": "TaxedMoneyRange",
        "start": Object {
          "__typename": "TaxedMoney",
          "gross": Object {
            "__typename": "Money",
            "amount": 24,
            "currency": "USD",
          },
          "net": Object {
            "__typename": "Money",
            "amount": 24,
            "currency": "USD",
          },
        },
        "stop": Object {
          "__typename": "TaxedMoney",
          "gross": Object {
            "__typename": "Money",
            "amount": 24,
            "currency": "USD",
          },
          "net": Object {
            "__typename": "Money",
            "amount": 24,
            "currency": "USD",
          },
        },
      },
      "priceRangeUndiscounted": Object {
        "__typename": "TaxedMoneyRange",
        "start": Object {
          "__typename": "TaxedMoney",
          "gross": Object {
            "__typename": "Money",
            "amount": 30,
            "currency": "USD",
          },
          "net": Object {
            "__typename": "Money",
            "amount": 30,
            "currency": "USD",
          },
        },
        "stop": Object {
          "__typename": "TaxedMoney",
          "gross": Object {
            "__typename": "Money",
            "amount": 30,
            "currency": "USD",
          },
          "net": Object {
            "__typename": "Money",
            "amount": 30,
            "currency": "USD",
          },
        },
      },
    },
    "thumbnail": Object {
      "__typename": "Image",
      "alt": "",
      "url": "http://localhost:8000/media/__sized__/products/saleordemoproduct_cl_bogo04_1_WF6PBIr-thumbnail-255x255.png",
    },
    "thumbnail2x": Object {
      "__typename": "Image",
      "url": "http://localhost:8000/media/__sized__/products/saleordemoproduct_cl_bogo04_1_WF6PBIr-thumbnail-510x510.png",
    },
  },
]
`;
